discord/api/util.go

```diff
@@ -0,0 +1,104 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/zond/diplicity/auth"
+	"github.com/zond/diplicity/game"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/v2/datastore"
+)
+
+func CreateAuthenticatedRequest(userId string, vars map[string]string, data any) (*GoaeoasRequest, error) {
+	log.Printf("api.CreateAuthenticatedRequest invoked - userId: %s; var: %s; data: %s \n", userId, vars, data)
+
+	user := createUserFromDiscordUserId(userId)
+	log.Printf("User instance created from Discord user ID: %+v\n", user)
+
+	bodyJson, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("Data marshalled to JSON: %s\n", bodyJson)
+
+	// Note, url and method don't matter because we skip router
+	httpRequest, err := http.NewRequest("GET", "", bytes.NewBuffer(bodyJson))
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("HTTP request created: %+v\n", httpRequest)
+
+	httpRequest.Header.Set("Content-Type", "application/json")
+
+	goaeoasRequest := &GoaeoasRequest{
+		req:  httpRequest,
+		vars: vars,
+		values: map[string]interface{}{
+			"user": user,
+		},
+	}
+	log.Printf("GoaeoasRequest created: %+v\n", goaeoasRequest)
+
+	_, err = getOrCreateUser(goaeoasRequest, user)
+	if err != nil {
+		return nil, err
+	}
+
+	return goaeoasRequest, nil
+}
+
+var NewGameDefaultValues = &game.Game{
+	Variant:                       "Classical",
+	PhaseLengthMinutes:            60 * 24,
+	NonMovementPhaseLengthMinutes: 60 * 24,
+	MaxHated:                      0,
+	MaxHater:                      0,
+	MinRating:                     0,
+	MaxRating:                     0,
+	MinReliability:                0,
+	MinQuickness:                  0,
+	Private:                       true,
+	NoMerge:                       false,
+	DisableConferenceChat:         true,
+	DisableGroupChat:              true,
+	DisablePrivateChat:            true,
+	NationAllocation:              0,
+	Anonymous:                     false,
+	LastYear:                      0,
+	SkipMuster:                    false,
+	ChatLanguageISO639_1:          "en",
+	GameMasterEnabled:             false,
+	RequireGameMasterInvitation:   false,
+}
```

Comment:
Note, for now all values in the newly created game are fixed. We can gradually add more customizability.